pkg/workloads/cortex/tf_api/api.py

```diff
@@ -252,38 +252,57 @@ def predict(deployment_name, api_name):
 
 
 def extract_signature(signature_def, signature_key):
-    if (
-        signature_def.get(signature_key) is None
-        or signature_def[signature_key].get("inputs") is None
-    ):
-        raise UserException(
-            'unable to find "' + signature_key + "\" in model's signature definition"
-        )
+    logger.info("signature defs found in model: {}".format(signature_def))
+
+    available_keys = list(signature_def.keys())
+    if len(available_keys) == 0:
+        raise UserException("unable to find signature defs in model")
+
+    if signature_key is None:
+        if len(available_keys) == 1:
+            logger.info(
+                "signature_key was not configured by user, using signature key '{}' found in signature def map".format(
+                    available_keys[0]
+                )
+            )
+            signature_key = available_keys[0]
+        else:
+            raise UserException(
+                "signature_key was not configured by user, please specify one the following keys '{}' found in signature def map".format(
+                    "', '".join(available_keys)
+                )
+            )
+    else:
+        if signature_def.get(signature_key) is None:
+            raise UserException(
+                "signature_key '{}' was not found, please specify one the following keys '{}' found in signature def map".format(
+                    signature_key, "', '".join(available_keys)
+                )
+            )
+
+    signature_def_val = signature_def.get(signature_key)
+
+    if signature_def_val.get("inputs") is None:
+        raise UserException("unable to find 'inputs' in signature def '{}'".format(signature_key))
 
     metadata = {}
-    for input_name, input_metadata in signature_def[signature_key]["inputs"].items():
+    for input_name, input_metadata in signature_def_val["inputs"].items():
         metadata[input_name] = {
             "shape": [int(dim["size"]) for dim in input_metadata["tensorShape"]["dim"]],
             "type": DTYPE_TO_TF_TYPE[input_metadata["dtype"]].name,
         }
-    return metadata
+    return signature_key, metadata
 
 
 @app.route("/<app_name>/<api_name>/signature", methods=["GET"])
 def get_signature(app_name, api_name):
-    ctx = local_cache["ctx"]
-    api = local_cache["api"]
-
     try:
-        metadata = extract_signature(
-            local_cache["metadata"]["signatureDef"],
-            local_cache["api"]["tf_serving"]["signature_key"],
-        )
+        signature = local_cache["signature"]
     except Exception as e:
```

Comment:
Do we still need the `try/except` here? In what case would an exception be thrown?